lib/src/serializable.dart

```diff
@@ -0,0 +1,239 @@
+// Copyright 2016 Workiva Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+library serializable_module.src.serializable;
+
+import 'dart:async';
+@MirrorsUsed(metaTargets: 'serializable_module.src.serializable.Reflectable')
+import 'dart:mirrors';
+
+import 'package:logging/logging.dart';
+import 'package:w_common/disposable.dart' show Disposable;
+import 'package:w_common/json_serializable.dart' show JsonSerializable;
+
+import 'package:w_module/src/event.dart';
+import 'package:w_module/src/module.dart';
+
+// Any classes / methods that are going to be reflected must annotate with this
+class Reflectable {
+  const Reflectable();
+}
+
+abstract class Bridge<T> extends Object with Disposable {
+  final Stream<Map> apiCallReceived;
+  Bridge(this.apiCallReceived);
+  void broadcastSerializedEvent(Map event);
+  void handleSerializedApiCall(T apiCall);
+
+  @override
+  void manageStreamSubscription(StreamSubscription subscription) {
+    super.manageStreamSubscription(subscription);
```

Comment:
This should probably be private and only used in this file, otherwise external consumers could tell the bridge to manage stream subscriptions that it shouldn't